lib/benchee/benchmark/collect/return_value.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Benchee.Benchmark.Collect.ReturnValue do
+  @moduledoc false
+
+  # Returns the value of the evaluated function.
+  # Used for pre checks.
+
+  # Does not strictly speaking implement Benchee.Benchmark.Collect because
+  # it does not return a metric.
+
+  @spec collect((-> any)) :: {any, any}
+  def collect(function) do
+    result = function |> Task.async() |> Task.await(:infinity)
+    {result, result}
```

Comment:
First I thought we don't need a new collector/it's odd but then I saw that we already implemented it with collectors and thought "ok maybe it's good". But, seeing here that we return the `result` twice made me question this again - because we already take the result and pass it on for the hooks! `Runner.collect/3` then returns the measurement but it has the return value of the function:

```elixir
  @doc """
  Takes one measure with the given collector.

  Correctly dispatches based on the number of iterations to perform.
  """
  def collect(
        scenario = %Scenario{function: function},
        scenario_context = %ScenarioContext{
          num_iterations: 1
        },
        collector
      ) do
    new_input = Hooks.run_before_each(scenario, scenario_context)
    function = main_function(function, new_input)

    {measurement, return_value} = invoke_collector(collector, function)

    Hooks.run_after_each(return_value, scenario, scenario_context)
    measurement
  end
```

Returning both seems like overhead, but maybe we could just implement a different `Runner.collect` function that in the end returns the `return_value` instead of the measurement? It also could be an option but that feels odd.

The more I think about it, I think just returning both here seems like a fun move (returning the measurement and the return value) and then we can do with it what we wish. 

Although... there is the case of multiple iterations in `Runner.collect` - which would be a real headache to deal with. In that case I believe doing a separate runner function for this purpose that only ever does one iteration (as is its purpose) works better.

Sorry I hope I make sense, just wanted to talk through the whole thought process with you :)

What do you think?

Reply:
> Returning both seems like overhead, but maybe we could just implement a different Runner.collect function that in the end returns the return_value instead of the measurement? It also could be an option but that feels odd.

Perhaps not so odd I think - it could be `Runner.run_once(scenario, scenario_context)` which doesn't even need to take in a collector.

> In that case I believe doing a separate runner function for this purpose that only ever does one iteration (as is its purpose) works better.

Agreed 👍

The only issue is that the new API would need to be bubbled up to `Benchee.Benchmark.RunOnce` as well. Perhaps RunOnce could be merged with Runner to simplify this, I'm not sure.